morphsnakes/cmorphsnakes.py

```diff
@@ -36,10 +37,15 @@ def step(self):
         """Perform a single step of the morphological Chan-Vese evolution."""
         self.cmorphacwe.step()
     
-    def run(self, iterations):
-        """Run several iterations of the morphological Chan-Vese method."""
-        for i in xrange(iterations):
+    def run(self, nb_iters):
+        """Run several nb_iters of the morphological Chan-Vese method."""
+        last_segm = np.array(self.levelset)
+        for i in range(nb_iters):
             self.step()
+            if np.array_equal(last_segm, self.levelset):
```

Comment:
The stopping criterion of morphsnakes is VERY tricky, and this is not the right way of doing it. One of the reasons is that it is not even consistent. For example, an odd number of the `smoothing` parameter ensures that the levelset at the next iteration will be different than the levelset at the current iteration in almost every case. This is not true with an even number of the `smoothing` parameter. This will make this stopping criterion to work sometimes, depending on the parameters. Given that this is the "official" implementation of the paper, I would like to avoid this kind of heuristics.